File: src/main/java/com/github/Doomsdayrs/Jikan4java/types/Main/User/Listing/AnimeList/AnimeListAnime.java
package com.github.Doomsdayrs.Jikan4java.types.Main.User.Listing.AnimeList;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.github.Doomsdayrs.Jikan4java.core.Retriever;
import com.github.Doomsdayrs.Jikan4java.types.Main.Anime.Anime;
import com.github.Doomsdayrs.Jikan4java.types.Main.Anime.Licensors;
import com.github.Doomsdayrs.Jikan4java.types.Main.Anime.Studios;

import java.util.ArrayList;
import java.util.concurrent.CompletableFuture;

/*
 * This file is part of Jikan4java.
 *
 * Jikan4java is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Jikan4java is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Jikan4java.  If not, see <https://www.gnu.org/licenses/>.
 * ====================================================================
 * Jikan4java
 * 03 / November / 2018
 *
 * @author github.com/doomsdayrs
 */
public class AnimeListAnime extends Retriever {

    @JsonProperty("mal_id")
    public int mal_id;
    @JsonProperty("title")
    public String title;
    @JsonProperty("video_url")
    public String video_url;
    @JsonProperty("url")
    public String url;
    @JsonProperty("image_url")
    public String image_url;
    @JsonProperty("type")
    public String type;
    @JsonProperty("watching_status")
    public int watching_status;
    @JsonProperty("score")
    public int score;
    @JsonProperty("watched_episodes")
    public int watched_episodes;
    @JsonProperty("total_episodes")
    public int total_episodes;
    @JsonProperty("airing_status")
    public int airing_status;
    @JsonProperty("season_name")
    public String season_name;
    @JsonProperty("season_year")
    public String season_year;
    @JsonProperty("has_episode_video")
    public boolean has_episode_video;
    @JsonProperty("has_promo_video")
    public boolean has_promo_video;
    @JsonProperty("has_video")
    public boolean has_video;
    @JsonProperty("is_rewatching")
    public boolean is_rewatching;
    @JsonProperty("tags")
    public String tags;
    @JsonProperty("rating")
    public String rating;
    @JsonProperty("start_date")
    public String start_date;
    @JsonProperty("end_date")
    public String end_date;
    @JsonProperty("watch_start_date")
    public String watch_start_date;
    @JsonProperty("watch_end_date")
    public String watch_end_date;
    @JsonProperty("days")
    public String days;
    @JsonProperty("storage")
    public String storage;
    @JsonProperty("priority")
    public String priority;
    @JsonProperty("added_to_list")
    public boolean added_to_list;
    @JsonProperty("studios")
    public ArrayList<Studios> studios;
    @JsonProperty("licensors")
    public ArrayList<Licensors> licensors;

    /**
     * Returns the Anime object of this object
     *
     * @return Anime Object
     */
    public CompletableFuture<Anime> getAnime() {
        return retrieve(Anime.class, baseURL + "/anime/" + mal_id);
    }

    @Override
    public String toString() {
        return "AnimeListAnime{" +
                "mal_id=" + mal_id +
                ", title='" + title + '\'' +
                ", video_url='" + video_url + '\'' +
                ", url='" + url + '\'' +
                ", image_url='" + image_url + '\'' +
                ", subType='" + type + '\'' +
                ", watching_status=" + watching_status +
                ", score=" + score +
                ", watched_episodes=" + watched_episodes +
                ", total_episodes=" + total_episodes +
                ", airing_status=" + airing_status +
                ", season_name='" + season_name + '\'' +
                ", season_year='" + season_year + '\'' +
                ", has_episode_video=" + has_episode_video +
                ", has_promo_video=" + has_promo_video +
                ", has_video=" + has_video +
                ", is_rewatching=" + is_rewatching +
                ", tags='" + tags + '\'' +
                ", rating='" + rating + '\'' +
                ", start_date='" + start_date + '\'' +
                ", end_date='" + end_date + '\'' +
                ", watch_start_date='" + watch_start_date + '\'' +
                ", watch_end_date='" + watch_end_date + '\'' +
                ", days='" + days + '\'' +
                ", storage='" + storage + '\'' +
                ", priority='" + priority + '\'' +
                ", added_to_list=" + added_to_list +
                ", studios=" + studios +
                ", licensors=" + licensors +
                '}';
    }
}

File: src/main/java/com/github/doomsdayrs/jikan4java/data/model/main/user/listing/animelist/AnimeListAnime.kt
package com.github.doomsdayrs.jikan4java.data.model.main.user.listing.animelist

import com.fasterxml.jackson.annotation.JsonProperty
import com.github.doomsdayrs.jikan4java.common.jikanURL
import com.github.doomsdayrs.jikan4java.core.Retriever
import com.github.doomsdayrs.jikan4java.data.base.*
import com.github.doomsdayrs.jikan4java.data.model.main.anime.Anime
import com.github.doomsdayrs.jikan4java.data.model.main.anime.Licensors
import com.github.doomsdayrs.jikan4java.data.model.main.anime.Studios
import java.util.*

/*
 * This file is part of Jikan4java.
 *
 * Jikan4java is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Jikan4java is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Jikan4java.  If not, see <https://www.gnu.org/licenses/>.
 * ====================================================================*/
/**
 * Jikan4java
 * 03 / November / 2018
 *
 * @author github.com/doomsdayrs
 */
data class AnimeListAnime(
		@JsonProperty("mal_id") override val malID: Int = 0,
		@JsonProperty("title") val title: String = "",
		@JsonProperty("video_url") val video_url: String = "",
		@JsonProperty("url") override val url: String,
		@JsonProperty("image_url") override val imageURL: String = "",
		@JsonProperty("type") override val type: String = "",
		@JsonProperty("watching_status") val watching_status: Int = 0,
		@JsonProperty("score") val score: Int = 0,
		@JsonProperty("watched_episodes") val watched_episodes: Int = 0,
		@JsonProperty("total_episodes") val total_episodes: Int = 0,
		@JsonProperty("airing_status") val airing_status: Int = 0,
		@JsonProperty("season_name") val season_name: String = "",
		@JsonProperty("season_year") val season_year: String = "",
		@JsonProperty("has_episode_video") val has_episode_video: Boolean = false,
		@JsonProperty("has_promo_video") val has_promo_video: Boolean = false,
		@JsonProperty("has_video") val has_video: Boolean = false,
		@JsonProperty("is_rewatching") val is_rewatching: Boolean = false,
		@JsonProperty("tags") val tags: String = "",
		@JsonProperty("rating") val rating: String = "",
		@JsonProperty("start_date") override val startDate: String = "",
		@JsonProperty("end_date") override val end_date: String = "",
		@JsonProperty("watch_start_date") val watch_start_date: String = "",
		@JsonProperty("watch_end_date") val watch_end_date: String = "",
		@JsonProperty("days") val days: String = "",
		@JsonProperty("storage") val storage: String = "",
		@JsonProperty("priority") val priority: String = "",
		@JsonProperty("added_to_list") val added_to_list: Boolean = false,
		@JsonProperty("studios") val studios: ArrayList<Studios>,
		@JsonProperty("licensors") val licensors: ArrayList<Licensors>
) : Retriever(), MyAnimeListID, MyAnimeListURL, MyAnimeListImageURL, MyAnimeListType, MyAnimeListStartEndDate {

	/**
	 * Returns the Anime object of this object
	 *
	 * @return Anime Object
	 */
	val anime by lazy { retrieve<Anime>("$jikanURL/anime/$malID") }
}